### PR TITLE
Fix creation properties

### PR DESCRIPTION
**Pull request by vespakoen**
They are all optional and I added some missing ones.

Cheers

### CODE REVIEW

**Comment by pravdomil** [comment]
hello, I made them all optional by using `Partial` type
```ts
(type: "button", bounds?: Bounds, text?: string, properties?: Partial<ControlPropertiesMap["button"]>): Button;
```

so there is no need make each property optional

**Comment by vespakoen** [comment]
Ahh I see, didn't notice! Reverted all the `?:`'s

File: shared/ScriptUI.d.ts [hunk 1]
```diff
@@ -2519,33 +2519,47 @@ interface ControlPropertiesMap {
 	/**
 	 * A Button object has no creation properties.
 	 * But the third argument to the add() method that creates it can be the initial text value.
+	 * @param name A unique name for the control. For a modal dialog, the special name "ok" makes this defaultElement, and the special name "cancel" makes this the cancelElement of the parent dialog.
```
Comment (pravdomil) [inline shared/ScriptUI.d.ts]: what about this param description
`Special name "ok" makes it primary button, and the special name "cancel" makes it cancel button for current dialog.`

**Comment by pravdomil** [comment]
here is a list of additional changes I have made
https://github.com/vespakoen/types-for-adobe/compare/541a43535ef94c19037d29734aaf78091d308c3b...pravdomil:3cdedea86b97ce87c9418857fef1260caa375e54

is it ok, can I merge it?

**Comment by vespakoen** [comment]
bitmoji

**Pull request merged**